File: src/main/java/com/thesilentnights/easylogin/service/task/CleanUp.java
package com.thesilentnights.easylogin.service.task;

import java.util.LinkedList;
import java.util.List;
import java.util.function.Consumer;

public class CleanUp implements Task {

    private final long delay;
    private final List<Consumer<Void>> cleanUpMethods = new LinkedList<>();
    private long tickCount = 0;

    public CleanUp(long delay) {
        this.delay = delay;
    }

    public void addCleanUpMethod(Consumer<Void> method) {
        if (method != null) {
            cleanUpMethods.add(method);
        }
    }

    @Override
    public void tick() {
        tickCount++;
        if (tickCount >= delay) {
            tickCount = 0;
            for (Consumer<Void> method : cleanUpMethods) {
                method.accept(null);
            }
        }
    }
}

File: src/main/java/com/thesilentnights/easylogin/service/task/CleanUp.kt
package com.thesilentnights.easylogin.service.task

import java.util.*
import java.util.function.Consumer


class CleanUp(var delay: Long) : Task {
    var tickCount: Long = 0
    var cleanUpMethods: LinkedList<Consumer<Void?>?> = LinkedList<Consumer<Void?>?>()

    fun addCleanUpMethods(method: Consumer<Void?>?) {
        cleanUpMethods.add(method)
    }

    override fun tick() {
        this.tickCount++
        if (this.tickCount >= this.delay) {
            this.tickCount = 0
            cleanUpMethods.forEach(Consumer { method: Consumer<Void?>? ->
                method!!.accept(null)
            })
        }
    }
}
